assets/scripts/streets/StreetMetaWidth.jsx

```diff
@@ -23,11 +23,41 @@ const DEFAULT_STREET_WIDTHS = [40, 60, 80]
 export class StreetMetaWidth extends React.Component {
   static propTypes = {
     intl: intlShape,
-    readOnly: PropTypes.bool,
+    editable: PropTypes.bool,
     street: PropTypes.object,
     updateStreetWidth: PropTypes.func
   }
 
+  static defaultProps = {
+    editable: true
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      isEditing: false
+    }
+
+    // Stores a ref to the street width <select> element
+    this.streetWidth = React.createRef()
+  }
+
+  /**
+   * If the `isEditing` state has toggled to true, show and display the <select>
+   */
+  componentDidUpdate = (prevProps, prevState) => {
+    if (this.state.isEditing === true && prevState.isEditing === false) {
+      this.streetWidth.current.focus()
+
+      window.setTimeout(() => {
```

Comment:
Can we have a comment here on why the timeout is needed?

Reply:
Oof. That would mean having to admit that I don't know why it's there or what it's supposed to be doing.

Usually a setTimeout of 0 means, "wait until the current JavaScript execution stack has finished, then run this code", or "do this immediately, but don't drop everything to do this right now, just do it as soon as you're ready." This can be vitally important when you're waiting for a browser to do a repaint, for example; you might need to read certain values from DOM, but you need to wait for a browser to update it first.

In this particular context I'm flummoxed because (1) I'm not sure what the timeout is waiting for, and (2) it _appears_ to want a click on a non-`<select>` element to render, focus, then automatically open the `<select>`. But it doesn't work that way now, and it doesn't work that way in this PR.

Given that this code has existed since 2013 and hasn't been revisited since, I'll just move to remove this code rather than attempt to document it.

Reply:
Yeah, I've had to use that hack in the past, and figured it was there for a reason but thought if we knew why now then we should document it. +1 for removing it

Reply:
(yeah, figured you knew the hack, but wanted to explain thoroughly for the benefit of bystanders!)